zebrad/src/components/mempool/tests.rs

```diff
@@ -94,5 +98,115 @@ async fn mempool_service_basic() -> Result<(), Report> {
 
     assert_eq!(rejected_ids, genesis_transaction_ids);
 
+    // Test `Request::Queue`
+    // Use the ID of the last transaction in the list
+    let txid = more_transactions.last().unwrap().id;
+    let response = service
+        .ready_and()
+        .await
+        .unwrap()
+        .call(Request::Queue(vec![txid.into()]))
+        .await
+        .unwrap();
+    let queued_responses = match response {
+        Response::Queued(queue_responses) => queue_responses,
+        _ => unreachable!("will never happen in this test"),
+    };
+    assert_eq!(queued_responses.len(), 1);
+    assert!(queued_responses[0].is_ok());
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn mempool_queue() -> Result<(), Report> {
+    // Using the mainnet for now
+    let network = Network::Mainnet;
+    let consensus_config = ConsensusConfig::default();
+    let state_config = StateConfig::ephemeral();
+    let (peer_set, _) = mock_peer_set();
+    let (sync_status, _recent_syncs) = SyncStatus::new();
+
+    let (state, _, _) = zebra_state::init(state_config, network);
+    let state_service = ServiceBuilder::new().buffer(1).service(state);
+    let (_chain_verifier, tx_verifier) =
+        zebra_consensus::chain::init(consensus_config.clone(), network, state_service.clone())
+            .await;
+
+    // Get transactions to use in the test
+    let (transactions_len, mut transactions) = unmined_transactions_in_blocks(10, network);
+    // We need at least 3 transactions for the test
+    assert!(transactions_len > 3);
+    // The first transaction to be added in the mempool which will be eventually
+    // put in the rejected list
+    let rejected_tx = transactions.first().unwrap().clone();
+    // A transaction not in the mempool that will be Queued
+    let new_tx = transactions.pop().unwrap();
+    // The last transaction that will be added in the mempool (and thus not rejected)
+    let stored_tx = transactions.last().unwrap().clone();
+
+    // Start the mempool service
+    let mut service = Mempool::new(
+        network,
+        peer_set,
+        state_service.clone(),
+        tx_verifier,
+        sync_status,
+    );
+    // Insert the genesis block coinbase transaction into the mempool storage.
+    service.storage.insert(rejected_tx.clone())?;
```

Comment:
This is only accessible with the `test` feature, right?

Reply:
Yes. In this particular case, it's because the test is a submodule of the mempool module